src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -236,8 +237,13 @@ private async Task HandleClusterTransaction(DocumentsOperationContext context, M
             if (topology.Promotables.Contains(ServerStore.NodeTag))
                 throw new DatabaseNotRelevantException("Cluster transaction can't be handled by a promotable node.");
 
-            var clusterTransactionCommand = new ClusterTransactionCommand(Database.Name, Database.IdentityPartsSeparator, topology, command.ParsedCommands, options, raftRequestId);
-            var result = await ServerStore.SendToLeaderAsync(clusterTransactionCommand);
+            using var cts = CreateHttpRequestBoundOperationToken();
+            var clusterTransactionCommand = new ClusterTransactionCommand(Database.Name, Database.IdentityPartsSeparator, topology, command.ParsedCommands, options, raftRequestId)
+            {
+                Timeout = Timeout.InfiniteTimeSpan // we rely on the http token to cancel the command
```

Comment:
What do you mean here?

Reply:
basically to have the same behavior as with a single node tx, you can cancel it only when passing a token to the `SaveChanges`.